### ISSUE TITLE
[SAE] K==1 assignment shortcut returns 1.0 unconditionally — wrong value for IBP / JumpReLU gates

### ISSUE THREAD

**Issue by SauersML** [SauersML/gam#597]
**Verified — PARTIAL/CONFIRMED.** Dimensionally safe (K=1 ⇒ length-1 assignment for all modes), but the *value* 1.0 is semantically wrong for the gated modes.

`try_assignments_row` (~`src/terms/sae_manifold.rs:1915-1917`):
```rust
if self.k_atoms() == 1 {
    return Ok(Array1::from_vec(vec![1.0]));
}
```
This short-circuits before the mode `match` (~1918-1929) and is not gated on `Softmax`.

- **Softmax, K=1:** 1.0 is correct (the only simplex point).
- **IBP, K=1, logit=0, temp=1, prior=1:** correct gate is sigmoid(0)=**0.5**, code returns 1.0.
- **JumpReLU, K=1, logit≤threshold:** correct gate is **0.0**, code returns 1.0.

Worse, the prior-gradient path only skips the softmax K==1 case, so with IBP/JumpReLU the optimizer can move a logit the reconstruction ignores — a free parameter affecting only the prior.

**Fix:** gate the shortcut on softmax only —
```rust
if self.k_atoms() == 1 && matches!(self.mode, AssignmentMode::Softmax { .. }) { ... }
```
and dispatch IBP/JumpReLU through their real assignment functions for K=1. Add tests: IBP K=1 logit=0 ⇒ 0.5; JumpReLU K=1 below threshold ⇒ 0.0.

_Source: static SAE/manifold-SAE audit; verified against current code at commit 3c1a4efa2._

**Comment by SauersML** [comment]
Fixed on main in `484cb83f3`: the K==1 shortcut is now gated to softmax only — `if self.k_atoms()==1 && matches!(self.mode, AssignmentMode::Softmax{..})` (sae_manifold.rs:1950); IBP/JumpReLU dispatch through their real assignment for K==1.
